build-image.sh

```diff
@@ -26,6 +26,10 @@ mount --bind $(pwd)/components $TARGET/building
 echo "Copying in some drive scripts"
 cp /usr/bin/qemu-arm-static $TARGET/usr/bin/qemu-arm-static
 cp pi-main.sh $TARGET/main.sh
+
+echo "Inserting version"
+git rev-parse HEAD > $TARGET/sb-version
```

Comment:
It'd be good to clarify what this is the version of in the name. Maybe call the file `sb-pi-image-version`?

Reply:
Possibly, but the idea is that from the version of `pi-image`, we'll know the version of everything else

Reply:
Ah, I see.

I was thinking mostly of the scenario where someone finds the file (in absence of knowing how the images are built) and wants to interpreted it.

IMO explicit is better than implicit, so I'd propose doing something like the following:
```
VERSION_FILE=$TARGET/sb-version
echo "Source Bots Raspberry Pi image" > $VERSION_FILE
echo >> $VERSION_FILE
git remote get-url origin >> $VERSION_FILE
git rev-parse HEAD >> $VERSION_FILE
echo >> $VERSION_FILE
git submodule foreach "git remote get-url origin; git rev-parse HEAD; echo" >> $VERSION_FILE
```

This has the handy result that it records the actual versions which were used, rather than merely those _expected_ to be used, meaning that local builds with modified version will still record usefully correct data.

The file would then look like this:
```
Source Bots Raspberry Pi image

git@github.com:sourcebots/pi-image
2c6bd16632e484e4309aede3b377cf3d4b70c9cd

Entering 'components/automount'
https://github.com/sourcebots/automount
9f87f806aef81e3ae5e4172dab1d63da15449f30

Entering 'components/robotd'
https://github.com/sourcebots/robotd.git
d77c37abf1dfd4baa4d6c05343a6097eca821b9f

Entering 'components/runusb'
https://github.com/sourcebots/runusb.git
5f73feec9d5be1130da1830640ed919651046db3

Entering 'components/sb-vision'
https://github.com/sourcebots/sb-vision.git
3844d77a630fe9126b5d360c6fd0bb53beb33a42
```